flink-yarn/src/main/java/org/apache/flink/yarn/AbstractYarnClusterDescriptor.java

```diff
@@ -545,6 +551,19 @@ protected YarnClusterClient deployInternal() throws Exception {
 		// Set-up ApplicationSubmissionContext for the application
 		ApplicationSubmissionContext appContext = yarnApplication.getApplicationSubmissionContext();
 
+		final ApplicationId appId = appContext.getApplicationId();
+
+		// ------------------ Add Zookeeper namespace to local flinkConfiguraton ------
+		String zkNamespace = getZookeeperNamespace();
+		// no user specified cli argument for namespace?
+		if(zkNamespace == null || zkNamespace.isEmpty()) {
+			// namespace defined in config? else use applicationId as default.
+			zkNamespace = flinkConfiguration.getString(ConfigConstants.ZOOKEEPER_NAMESPACE_KEY, String.valueOf(appId));
+			setZookeeperNamespace(zkNamespace);
+		}
+
+		flinkConfiguration.setString(ConfigConstants.ZOOKEEPER_NAMESPACE_KEY, zkNamespace);
```

Comment:
this should be in an `else` block, otherwise we may set a property even though it is already set.